tech-spec.md

```diff
@@ -0,0 +1,310 @@
+# Tech spec
+
+The ODM sharing library aims to be able to read the most used file formats and
+databases, provide useful feedback to help build correct filters, and output
+easy-to-share result data.
+
+## Overview
+
+A sharing schema (in CSV format) is used to define individual rules (as rows),
+which come together to form a data query.
+
+The query rules are parsed into an abstract syntax tree (AST), which represents
```

Comment:
it's a good suggestion, but it would imply that the AST only exist to separate those two steps, however, I see it more as a necessary step in the process which also makes it modular as a bonus.

Reply:
Its not really necessary in that the SQL query can be generated directly from the rules sheet right? It is necessary in order to keep the code modular. In any case, its a minor comment, so I'll resolve the conversation.